processor/k8sprocessor/doc.go

```diff
@@ -16,12 +16,26 @@
 //
 // The processor automatically discovers k8s resources (pods), extracts metadata from them and adds the
 // extracted metadata to the relevant spans, metrics and logs. The processor use the kubernetes API to discover all pods
-// running in a cluster, keeps a record of their IP addresses and interesting metadata. Upon receiving telemetry data,
-// the processor looks for presence of well-known resource attributes which might contain IP address ("ip",
-// "k8s.pod.ip" for logs, metrics or traces and "host.name" for metrics). If this field is not available, or it
-// does not contain a valid IP address, the processor tries to identify the source IP address of the service
-// that sent the telemetry data.
-// If a match is found, the cached metadata is added to the data as resource attributes.
+// running in a cluster, keeps a record of their IP addresses, pod UIDs and interesting metadata.
+// The rules for associating the source record with specific Pod Metadata are configured via "pod_association" key.
+// It represents a list of rules that are executed in order until the first one is able to do the match.
+// Each rule is specified as a pair of from (representing the rule type) and name (representing the extracted key name).
+// Following rule types are available:
+// 	  from: "labels" - takes the attribute from resource attributes (the value can contain either IP address or Pod UID)
+// 	  from: "connection" - takes the IP attribute from connection context (if available) and automatically
+// 						   associates it with "k8s.pod.ip" attribute
```

Comment:
Mixed tabs and spaces:

```suggestion
// Following rule types are available:
//   from: "labels" - takes the attribute from resource attributes (the value can contain either IP address or Pod UID)
//   from: "connection" - takes the IP attribute from connection context (if available) and automatically
//     associates it with "k8s.pod.ip" attribute
```